### DISCUSSION TITLE
树状结构趋势表已经手动展开的结构树，变成初始化全部收起状态

### DISCUSSION THREAD

**Discussion by Hongbo-Sheng** [antvis/S2#2831] [Q&A] [answered]
诉求是手动展开过的结构树，在点击按钮数据重新加载时候回到默认全部收起状态，
![image](https://github.com/user-attachments/assets/36016602-4c01-4918-9e40-c9d4f2b35f01)

用了文档提供的SpreadSheet和Interaction方法没什么效果，
![image](https://github.com/user-attachments/assets/f8a02605-2935-40f1-bfec-bc5f6325f963)
用的是SheetComponent，引入方式import { SheetComponent } from '@antv/s2-react';
![image](https://github.com/user-attachments/assets/11e65983-f260-4dea-b28e-b4331c28f68b)
containerRef.current.interaction.spreadsheet.destroy()可以销毁，然后reset或者其他一些都没效果这是为什么？这里有什么限制吗？求大佬help

**Comment by lijinke666** [comment]
用的什么版本, 多给点有效信息应该不难吧?

> 然后reset或者其他一些都没效果这是为什么？这里有什么限制吗？求大佬help



你框出来的是 2.x 新增的 api, 你不会是看的 2.x 的文档, 然后用的 1.x 代码吧? api 在什么地方调用的, 什么版本, 一概不知, 没人知道为什么.

**Comment by lijinke666** [comment, marked as answer]
手动折叠展开的信息, 会保存到 `s2.options.style` 中, 在监听数据变更后 (比如 useEffect) 使用 `s2.setOptions({ style: { xxx: null } })` 重置清空即可

不同版本名字可能不同, 自行对照文档

> 1.x

![image](https://github.com/user-attachments/assets/de23a730-df79-446d-bb6c-cfd36403750b)

> 2.x

![image](https://github.com/user-attachments/assets/e796a642-06e5-4e78-91cb-2933f6854055)

**Comment by Hongbo-Sheng** [comment]
好的谢谢大佬了
